src/main/java/baseball/domain/game/BaseballGame.java

```diff
@@ -0,0 +1,48 @@
+package baseball.domain.game;
+
+import baseball.domain.game.result.PlayResult;
+
+import java.util.List;
+
+public class BaseballGame {
+    private Balls computerBalls;
+    private GameState gameState;
+
+    public BaseballGame() {
+        initializeData();
+    }
+
+    private void initializeData() {
+        this.computerBalls = new Balls(createRandomNumbers());
+        this.gameState = GameState.PLAYING;
+    }
+
+    protected List<Integer> createRandomNumbers() {
+        return RandomNumberGenerator.createRandomNumbers();
+    }
+
+    public PlayResult play(Balls userBalls) {
+        PlayResult playResult = PlayResult.from(computerBalls.compare(userBalls));
+
+        if (playResult.isThreeStrike()) {
+            gameState = gameState.end();
+        }
+        return playResult;
+    }
```

Comment:
GameState는 Enum이라서 충분히 아래처럼 가능 했을텐데 GameState한테 메시지를 보낸 이유가 있을까요?
```suggestion
    public PlayResult play(Balls userBalls) {
        PlayResult playResult = PlayResult.from(computerBalls.compare(userBalls));

        if (playResult.isThreeStrike()) {
            gameState = GameState.END;
        }
        return playResult;
    }
```

Reply:
객체 필드에 직접 접근하지 않도록 하기 위함이였어요!
리팩토링 과정에서 객체에 메시지를 보낼 수 있는 부분들을 수정하였고, 말씀주신 부분도 그 중 하나였습니다.!

Reply:
전 그래도 enum클래스인지라 `gameState = GameState.END;` 이게 더 이해하기 쉬워보입니다..!